Add I18N support

jamadden opened the pull request:
Fixes #5.

jamadden committed:
Fit the webapp into the localization model used by the server (just tweak the locations of some files, prepping the way for locale-specific icons too), and take over serving strings.js based on the negotiated locale. Currently this is only using HTTP Accept-Language, preference and/or cookie support coming soon. Needs a test case, but empiracally working.

jamadden committed:
Support for query params, cookies, and user preferences for locale selection. unit tests.

jamadden committed:
More I18N: hook up pyramid's locale negotiator and translation directories to existing infrastructure.

jamadden committed:
Safety for debugtoolbar

papachoco committed:
fix method call; clean up

papachoco committed:
clean up

papachoco committed:
use nti.property

cutz committed:
we always want our locale negotiator registered (not just when we pull in the webapp locale views). load the nti.app.i18n config by default but pull the view scan out if for inclusion in the buildout included file.  Fixes currency formatting in oubound.

jzuech3 committed:
Update Lazy imports

papachoco committed:
pep8;pylint; remove unicode literals

jamadden committed:
Decouple from private implementation details and fix tests.

jamadden committed:
Remove unused imports

cutz commented on src/nti/app/pyramid_zope/i18n/__init__.py:
```diff
@@ -0,0 +1,68 @@
+to have the right setting, either for testing, or to support broken
+browsers, or to explicitly enable user preferences. Supporting user
+preferences is easy, an implementation of mutable preferences is
+provided for :class:`.IUser` objects. For temporary testing or for the
```
Is the `IUser` bit still accurate?

jamadden replied:
No. And I can't find where it ever would have been 😿  (We'd need an implementation of `IUserPreferredLanguages` or `IUserModifiablePreferredLanguages` adapting the IUser — now the IPrincipal.)

jamadden committed:
pylint found duplicate header lines.

Which it only does when scanning the whole project, not when editing a file. sigh.

jamadden commented:
Note that nti.appserver:configure.zcml will have manual conflicting directives from zope.publisher that have to be removed after this. (We manually registered just two of its directives. I don't know why we didn't just load the whole configuration; there's a comment to that effect.)

jamadden committed:
Tweak comment about preferences [skip ci]
